lib/Core/MemoryManager.cpp

```diff
@@ -40,7 +98,31 @@ MemoryObject *MemoryManager::allocate(uint64_t size, bool isLocal,
     klee_warning_once(0, "Large alloc: %lu bytes.  KLEE may run out of memory.",
                       size);
 
+  // Return NULL if size is zero, this is equal to error during allocation
+  if (NullOnZeroMalloc && size == 0)
+    return 0;
+
   uint64_t address = 0;
+  if (DeterministicAllocation) {
+    if ((alignment & -alignment) != alignment) {
+      klee_warning("Only alignment of power of two is supported");
+      return 0;
+    }
+
+    address = ((uint64_t)nextFreeSlot + alignment - 1) & -alignment;
+
+    size_t alloc_size = std::max(size, (uint64_t)1);
```

Comment:
This happens quite often (e.g. vararg handling). It's part of the POSIX standard. Same for `free()` which can take a NULL ptr or a ptr pointing to an allocation of size 0.
As a programmer calling stdlibc, it's useful as you don't need to explicitly check for the 0 case.